server/utils/scores.ts

```diff
@@ -1,150 +1,129 @@
-    validatorInherents.set(epoch, { rewarded: accRewarded + Math.max(rewarded, 0), missed: accMissed + Math.max(missed, 0) })
+  // Process epochs in order and handle missing data
+  for (let epoch = range.fromEpoch; epoch <= range.toEpoch; epoch++) {
+    const data = epochData.get(epoch) || { missed: -1, rewarded: -1 }
+    inherentsPerEpoch.set(epoch, data)
+    activeEpochStates.push(data.missed === 0 && data.rewarded === 0 ? 0 : 1)
   }
 
-  const scores = Array.from(validatorsParams.entries()).map(([validatorId, { inherentsPerEpoch }]) => {
-    const activeEpochStates = Array.from({ length: range.toEpoch - range.fromEpoch + 1 }, (_, i) => {
-      if (!inherentsPerEpoch.has(range.fromEpoch + i))
-        return 0
-      const { rewarded, missed } = inherentsPerEpoch.get(range.fromEpoch + i)!
-      if (rewarded === 0 && missed === 0)
-        return 0
-      return 1
-    })
-    const { dominanceRatioViaBalance, dominanceRatioViaSlots } = dominanceLastEpochByValidator.get(validatorId) ?? emptyDominance
-    const dominance: ScoreParams['dominance'] = { dominanceRatio: dominanceRatioViaBalance === -1 ? dominanceRatioViaSlots : dominanceRatioViaBalance }
-    const availability: ScoreParams['availability'] = { activeEpochStates }
-    const reliability: ScoreParams['reliability'] = { inherentsPerEpoch }
-
-    const reason = {
-      missedEpochs: activeEpochStates.map((s, i) => s === 0 ? range.fromEpoch + i : -1).filter(e => e !== -1),
-      goodSlots: Array.from(inherentsPerEpoch.values()).reduce((acc, { rewarded }) => acc + rewarded, 0),
-      badSlots: Array.from(inherentsPerEpoch.values()).reduce((acc, { missed }) => acc + missed, 0),
-    }
-
-    const score = computeScore({ availability, dominance, reliability })
-    const newScore: NewScore = { validatorId: Number(validatorId), epochNumber: range.toEpoch, ...score, reason }
-    return newScore
-  })
-
-  // If the range is the default window dominance or the range starts at the PoS fork block, we persist the scores
-  // TODO Once the chain is older than 9 months, we should remove range.fromBlockNumber === 1
-  if (range.toEpoch - range.fromEpoch + 1 === DEFAULT_WINDOW_IN_EPOCHS || range.fromEpoch === 1)
-    await persistScores(scores)
-
-  const { data: validators, error: errorValidators } = await fetchValidatorsScoreByIds(scores.map(s => s.validatorId))
-  if (errorValidators || !validators)
-    return { error: errorValidators, data: undefined }
-  return { data: { validators, range }, error: undefined }
+  const params: ScoreParams = {
+    availability: { activeEpochStates },
+    dominance: { dominanceRatio },
+    reliability: { inherentsPerEpoch },
+  }
+  const [scoreAreOk, errorScore, scoreValues] = computeScore(params)
+  if (!scoreAreOk)
+    return [false, errorScore, undefined]
+  const score: Score = {
+    ...scoreValues,
+    validatorId,
+    epochNumber: range.toEpoch,
+    reason: '{}',
```

Comment:
This is always `'{}'`, it is never replaced by a real value. Example [here](https://9e0a0585.validators-api-mainnet.pages.dev/api/v1/validators/NQ38%20VK34%20DRBL%20S3CN%20M9KM%208UJN%209JY2%202KFN%20VQQH)

![image](https://github.com/user-attachments/assets/80aa4e56-732e-4f18-8d3c-ce446af495ec)

Reply:
fyi I am removing `reason` completely.

In the future we can explore this more, but for now we don't use it